index.d.ts

```diff
@@ -5,10 +5,41 @@ export class Vec3 {
   y: number;
   z: number;
 
+  /**
+   * Returns true when it is a zero vector.
+   */
+  isZero(): boolean;
+
+  /**
+   * Access component by index
+   */
+  at(id: number): number;
+
+  /**
+   * Returns an array component x, z
+   */
+  xz(): [number, number];
```

Comment:
Is that useful?

[p.x, p.z] seems clearer to use

Reply:
I think so, please consider a longer name like `bot.entity.position`.

It can be more useful if we accept more constructors like `new Vec3(v.xz(), v.y)` meaning `new Vec3(v.x, v.z, v.y)`, which is very similar to what we do in GLSL.